pkg/collector/corechecks/containers/containerd/containerd.go

```diff
@@ -270,16 +268,31 @@ func isExcluded(ctn containers.Container, fil *ddContainers.Filter) bool {
 	return fil.IsExcluded("", ctn.Image, ctn.Labels["io.kubernetes.pod.namespace"])
 }
 
-func convertTasktoMetrics(metricTask *containerdTypes.Metric) (*v1.Metrics, error) {
-	metricAny, err := typeurl.UnmarshalAny(&types.Any{
-		TypeUrl: metricTask.Data.TypeUrl,
-		Value:   metricTask.Data.Value,
-	})
+func computeLinuxSpecificMetrics(metrics *v1.Metrics, sender aggregator.Sender, cu cutil.ContainerdItf, ctn containerd.Container, createdAt time.Time, currentTime time.Time, tags []string) {
+	computeMem(sender, metrics.Memory, tags)
+
+	ociSpec, err := cu.Spec(ctn)
 	if err != nil {
-		log.Errorf(err.Error())
-		return nil, err
+		log.Warnf("Could not retrieve OCI Spec from: %s: %v", ctn.ID(), err)
+	}
+
+	var cpuLimits *specs.LinuxCPU
+	if ociSpec != nil && ociSpec.Linux != nil && ociSpec.Linux.Resources != nil {
+		cpuLimits = ociSpec.Linux.Resources.CPU
+	}
+	computeCPU(sender, metrics.CPU, cpuLimits, createdAt, currentTime, tags)
+
+	if metrics.Blkio.Size() > 0 {
+		computeBlkio(sender, metrics.Blkio, tags)
+	}
+
+	if len(metrics.Hugetlb) > 0 {
+		computeHugetlb(sender, metrics.Hugetlb, tags)
 	}
-	return metricAny.(*v1.Metrics), nil
+}
+
+func computeWindowsSpecificMetrics(windowsStats *wstats.Statistics) {
+	// TODO: Should report similar information to computeLinuxSpecificMetrics().
```

Comment:
My original plan was to add that on a separate PR, but we can add it here if you prefer. @vboulineau asked for this too in a comment above.

Reply:
@clamoriniere , @vboulineau : I added a new commit with the windows metrics:

- CPU (reporting some of the metrics already reported in Linux): `containerd.cpu.total`, `containerd.cpu.user`, `containerd.cpu.system`. (rates)
- Memory (reporting new metrics because I didn't find an equivalent in the ones already reported): `containerd.mem.commit`, `containerd.mem.commit_peak`, `containerd.mem.private_working_set`. (gauges)
- Storage (reporting new metrics also): `containerd.storage.read`, `containerd.storage.write`. (rates)

All the names are based on the fields present in the `wstats.Statistics` struct used in the `computeWindowsSpecificMetrics` function.